services/gitdiff/gitdiff.go

```diff
@@ -1147,16 +1147,20 @@ func GetDiff(gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff
 	}
 	diff.Start = opts.SkipTo
 
-	checker, deferable := gitRepo.CheckAttributeReader(opts.AfterCommitID)
+	checker, deferable, err := gitRepo.CheckAttributeReader(opts.AfterCommitID)
+	if err != nil {
+		return nil, err
+	}
 	defer deferable()
 
 	for _, diffFile := range diff.Files {
-
 		gotVendor := false
 		gotGenerated := false
 		if checker != nil {
 			attrs, err := checker.CheckPath(diffFile.Name)
-			if err == nil {
+			if err != nil {
+				log.Error("CheckPath returns error: %v", err)
+			} else {
 				if vendored, has := attrs["linguist-vendored"]; has {
```

Comment:
Wait, we have the same code in two different files?
Should we perhaps extract the common things into one method?